test_vectors/README.md

```diff
@@ -1,9 +1,86 @@
-## Test vectors
+# Test vectors
 
- - [Tree math](https://github.com/mlswg/mls-implementations/blob/master/test_vectors/treemath.md)
+This directory contains binary files with pre-serialized test
+vectors.  Each file contains a single structure in a TLS-syntax
+format defined below (in each case, the `XXXTestVectors` struct).
 
- - [UserInitKey](https://github.com/mlswg/mls-implementations/blob/master/test_vectors/userinitkey.md)
+## Tree Math
 
-- [ECIES](https://github.com/mlswg/mls-implementations/blob/master/test_vectors/ecies.md)
+File: [tree_math.bin](https://github.com/mlswg/mls-implementations/blob/master/test_vectors/tree_math.bin)
 
-- [Trees and nodes](https://github.com/mlswg/mls-implementations/blob/master/test_vectors/treesnodes.md)
+```
+struct {
+  uint32 root<0..2^32-1>;
+  uint32 left<0..2^32-1>;
+  uint32 right<0..2^32-1>;
+  uint32 parent<0..2^32-1>;
+  uint32 sibling<0..2^32-1>;
+} TreeMathTestVectors;
+```
+
+These vectors have the following meaning, where the tree relations
+are as defined in [the specification](https://github.com/mlswg/mls-protocol/blob/master/draft-ietf-mls-protocol.md#tree-computation-terminology):
+
+* `root[i]` is the index of the root of a tree with `i+1` leaves
+* The remaining vectors are all within the context of a tree with
+  255 leaves:
```

Comment:
TBH, because that's what @raphaelrobert did in his first pass :)  But yeah, the unbalancing seems useful, since it will hit some of the unusual cases.

Reply:
Right, 255 was for "historical reasons". I think @bifurcation has a point and we should generate several test vectors for wildly unbalanced trees in addition to balanced ones.

Reply:
FWIW, 255 is plenty unbalanced for me!